selfhost/typechecker.jakt

```diff
@@ -5257,6 +5257,70 @@ struct Typechecker {
         return None
     }
 
+    function typecheck_format(mut this, args: [(String, CheckedExpression)]) throws {
+        guard args[0].1 is QuotedString(val: format_string, span) else{
+            eprintln("Internal Compiler Error: `typecheck_format` called with dynamic format string: `{}`", args[0].1)
+            abort()
+        }
+        
+        mut n_args = 1uz
+        mut index = 0uz
+        mut in_format_specifier = false
+
+        while index < format_string.length() {
+            let current_span = Span(file_id: span.file_id, start: span.start + index + 1, end: span.start + index + 1)
+            if in_format_specifier {
+                guard format_string.byte_at(index) != b'{' else {
+                    .error("Found '{' inside of format specifier", current_span)
+                    return
+                }
+
+                // FIXME: check for valid formatting specifiers, like ':x'
```

Comment:
Also add a fixme for explicit indices like `"{1}"`, I'd suggest looking at AK's comptime checker, that's been tested enough to be reasonably correct